### ISSUE TITLE
[Bug]: Invalid mimetypelist.js hash

### ISSUE THREAD

**Issue by idressos** [nextcloud/server#44315]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After running occ maintenance:mimetype:update-js, core integrity checks fail with the following error:

  - INVALID_HASH:
    - core/js/mimetypelist.js:
      - expected: 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201
      - current: 24e6faa4a1361e51ff70027024f41c4b37fdb5f6f347ceb095450ddbde2d0a6b6dd1403be92a912d2889064a9736a9bed89d7dc62470035fc1299d4b977c5f46

The mimetypelist.js file has not been altered manually in any way.

### Steps to reproduce

1. In your server's command line, navigate to the Nextcloud root installation directory
2. Execute php occ maintenance:mimetype:update-js
3. Execute php occ integrity:check-core, or wait for cron task to do it on scheduled interval

### Expected behavior

No integrity errors.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

master

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "records.aviet.aero"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.3.2",
        "overwrite.cli.url": "http:\/\/records.aviet.aero",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "dbindex": 0
        },
        "theme": "",
        "loglevel": 3,
        "default_phone_region": "MT",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "data-fingerprint": "b3066f76aae76bf77dd2a53fa6a92427",
        "defaultapp": "files",
        "maintenance_window_start": 1
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - analytics: 4.12.0
  - announcementcenter: 6.8.0
  - appointments: 2.0.4
  - approval: 1.2.0
  - audioplayer: 3.4.1
  - bruteforcesettings: 2.8.0
  - calendar: 4.6.6
  - certificate24: 0.3.0
  - checksum: 1.2.3
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - data_request: 1.15.0
  - dav: 1.29.1
  - deck: 1.12.2
  - encryption: 2.16.0
  - end_to_end_encryption: 1.14.3
  - event_update_notification: 2.3.0
  - external: 5.3.1
  - federatedfilesharing: 1.18.0
  - files: 2.0.0
  - files_accesscontrol: 1.18.0
  - files_automatedtagging: 1.18.0
  - files_bpm: 1.3.0
  - files_lock: 28.0.2
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_retention: 1.17.1
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - forms: 4.1.1
  - groupfolders: 16.0.4
  - groupquota: 0.1.12
  - impersonate: 1.15.0
  - integration_openai: 1.2.1
  - llm: 1.2.1
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - notes: 4.9.2
  - notifications: 2.16.0
  - notify_push: 0.6.9
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - quota_warning: 1.18.0
  - recognize: 6.1.1
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - richdocuments: 8.3.2
  - richdocumentscode: 23.5.904
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - side_menu: 3.11.8
  - support: 1.11.0
  - suspicious_login: 6.0.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - theming_customcss: 1.15.0
  - translate: 2.1.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - federation: 1.18.0 (installed 1.15.0)
  - files_external: 1.20.0 (installed 1.19.0)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - metadata: 0.19.0 (installed 0.19.0)
  - nextcloud_announcements: 1.17.0 (installed 1.17.0)
  - survey_client: 1.16.0 (installed 1.16.0)
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0
```

### Nextcloud Signing status

```shell
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201
                            [current] => 24e6faa4a1361e51ff70027024f41c4b37fdb5f6f347ceb095450ddbde2d0a6b6dd1403be92a912d2889064a9736a9bed89d7dc62470035fc1299d4b977c5f46
                        )

                )

        )

)
```

**Comment by idressos** [comment]
@Matthias-Ab

**Comment by SjoerdV** [comment]
Same issue here, I think in my case the culprit is the 'drawio' app that added a mimetype entry while not being built for the new integrity check system. Might just have to wait on app updates to conform to the new regime?

**Comment by flotpg** [comment]
https://github.com/nextcloud/server/issues/26022#issuecomment-2063476387

Same issue here on Nextcloud 28.0.4.

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201
                            [current] => 5b121122c0bec0580402c04605700056ee06eb553a4692bfc085ad3fc17aa49af43ca7795e98cbd22383020bfa9809bf3cc824b1673595f461479afb602bcec4
                        )

                )

        )

)
```

It looks like the released zip file for 28.0.4 contains mimetypelist.js with the expected sha512 hash:

https://github.com/nextcloud/server/archive/refs/tags/v28.0.4.zip
550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201

But the one from the repo has a different hash:

https://github.com/nextcloud/server/blob/master/core/js/mimetypelist.js
https://raw.githubusercontent.com/nextcloud/server/master/core/js/mimetypelist.js
5b121122c0bec0580402c04605700056ee06eb553a4692bfc085ad3fc17aa49af43ca7795e98cbd22383020bfa9809bf3cc824b1673595f461479afb602bcec4
I'm took the file from the released zip file and it fixed the issue for me! ✅

**Comment by joshtrichards** [comment]
https://github.com/nextcloud/server/blob/4af9bef7121c91359c537c42b8bf5c71fdb37e79/lib/private/IntegrityCheck/Checker.php#L170-L178

Do have any MIME type aliases defined?

https://docs.nextcloud.com/server/latest/admin_manual/configuration_mimetypes/index.html#mimetype-aliases

Generally via `config/mimetypealiases.json`

**Comment by nfk** [comment]
Why extra mimetype are not loaded from `mimetypealiases.json`? It is weird to patch source code on a deployed product  instead of reading an extra config file.
In my way `core/js/mimetypelist.js` file is updated on `maintenance:theme:update` command which calls the parent command indirectly `maintenance:mimetype:update-js`.


```diff
root@8560f7d49296:/var/www/html# diff /usr/src/nextcloud/core/js/mimetypelist.js ./core/js/mimetypelist.js                                           
79d78
<     "application/x-compressed": "package/x-generic",
153c152,177
<       themes: []
---
>       themes: {
>     "eCloud": [
>         "android-package",
>         "application",
>         "application-pdf",
>         "audio",
>         "file",
>         "folder",
>         "folder-archive",
>         "folder-drag-accept",
>         "folder-encrypted",
>         "folder-external",
>         "folder-public",
>         "folder-shared",
>         "folder-starred",
>         "image",
>         "picture_as_pdf",
>         "slideshow",
>         "text",
>         "text-vcard",
>         "video",
>         "x-office-document",
>         "x-office-presentation",
>         "x-office-spreadsheet"
>     ]
> }
```

So if there is no way to load extra mimetype list. The file  `core/js/mimetypelist.js` cannot be part of integrity check. If anyone has a workaround please let me know.

**Comment by nfk** [comment]
So 2 issues on Nextcloud 29.

1. the commit c44656aa0a617c7ec91a10389794c947f90be018 removed a mimetype entry but the `core/js/mimetypelist.js` file had not been regenerated. So integrity check is failing.
2. if you use  a custom theme with file types the custom hack from integrity check does not work because theme file type are added. https://github.com/nextcloud/server/blob/7543679dd2a8fa4607f0558471a1110938b4fd0a/lib/private/IntegrityCheck/Checker.php#L175

So I advice to remove the integrity check on a generated file it does make sense. https://github.com/nextcloud/server/blob/7543679dd2a8fa4607f0558471a1110938b4fd0a/core/js/mimetypelist.js#L2

Otherwise I have a potential fix for NC 29 see below.

```diff
diff --git a/core/Command/Maintenance/Mimetype/GenerateMimetypeFileBuilder.php b/core/Command/Maintenance/Mimetype/GenerateMimetypeFileBuilder.php
index db2f3e44f1c..69de9d052fe 100644
--- a/core/Command/Maintenance/Mimetype/GenerateMimetypeFileBuilder.php
+++ b/core/Command/Maintenance/Mimetype/GenerateMimetypeFileBuilder.php
@@ -34,7 +34,7 @@ class GenerateMimetypeFileBuilder {
         * @param array $aliases
         * @return string
         */
-       public function generateFile(array $aliases): string {
+       public function generateFile(array $aliases, bool $includeThemes = true): string {
                // Remove comments
                $aliases = array_filter($aliases, static function ($key) {
                        // Single digit extensions will be treated as integers
@@ -61,33 +61,35 @@ class GenerateMimetypeFileBuilder {
 
                // Fetch all themes!
                $themes = [];
-               $dirs = new \DirectoryIterator(\OC::$SERVERROOT.'/themes/');
-               foreach ($dirs as $dir) {
-                       //Valid theme dir
-                       if ($dir->isFile() || $dir->isDot()) {
-                               continue;
-                       }
+               if ($includeThemes) {
+                       $dirs = new \DirectoryIterator(\OC::$SERVERROOT.'/themes/');
+                       foreach ($dirs as $dir) {
+                               //Valid theme dir
+                               if ($dir->isFile() || $dir->isDot()) {
+                                       continue;
+                               }
 
-                       $theme = $dir->getFilename();
-                       $themeDir = $dir->getPath() . '/' . $theme . '/core/img/filetypes/';
-                       // Check if this theme has its own filetype icons
-                       if (!file_exists($themeDir)) {
-                               continue;
-                       }
+                               $theme = $dir->getFilename();
+                               $themeDir = $dir->getPath() . '/' . $theme . '/core/img/filetypes/';
+                               // Check if this theme has its own filetype icons
+                               if (!file_exists($themeDir)) {
+                                       continue;
+                               }
 
-                       $themes[$theme] = [];
-                       // Fetch all the theme icons!
-                       $themeIt = new \DirectoryIterator($themeDir);
-                       foreach ($themeIt as $fileInfo) {
-                               if ($fileInfo->isFile()) {
-                                       $file = preg_replace('/.[^.]*$/', '', $fileInfo->getFilename());
-                                       $themes[$theme][] = $file;
+                               $themes[$theme] = [];
+                               // Fetch all the theme icons!
+                               $themeIt = new \DirectoryIterator($themeDir);
+                               foreach ($themeIt as $fileInfo) {
+                                       if ($fileInfo->isFile()) {
+                                               $file = preg_replace('/.[^.]*$/', '', $fileInfo->getFilename());
+                                               $themes[$theme][] = $file;
+                                       }
                                }
-                       }
 
-                       //Remove Duplicates
-                       $themes[$theme] = array_values(array_unique($themes[$theme]));
-                       sort($themes[$theme]);
+                               //Remove Duplicates
+                               $themes[$theme] = array_values(array_unique($themes[$theme]));
+                               sort($themes[$theme]);
+                       }
                }
 
                //Generate the JS
diff --git a/core/js/mimetypelist.js b/core/js/mimetypelist.js
index 3becf6e37d3..ab1ad80e142 100644
--- a/core/js/mimetypelist.js
+++ b/core/js/mimetypelist.js
@@ -76,7 +76,6 @@ OC.MimeTypeList={
     "application/x-7z-compressed": "package/x-generic",
     "application/x-bzip2": "package/x-generic",
     "application/x-cbr": "text",
-    "application/x-compressed": "package/x-generic",
     "application/x-dcraw": "image",
     "application/x-deb": "package/x-generic",
     "application/x-fictionbook+xml": "text",
diff --git a/lib/private/IntegrityCheck/Checker.php b/lib/private/IntegrityCheck/Checker.php
index 32749a2c5f1..93e7ea33c1e 100644
--- a/lib/private/IntegrityCheck/Checker.php
+++ b/lib/private/IntegrityCheck/Checker.php
@@ -172,7 +172,7 @@ class Checker {
                                $newFile = $oldMimetypeList->generateFile($this->mimeTypeDetector->getAllAliases());
                                $oldFile = $this->fileAccessHelper->file_get_contents($filename);
                                if ($newFile === $oldFile) {
-                                       $hashes[$relativeFileName] = hash('sha512', $oldMimetypeList->generateFile($this->mimeTypeDetector->getOnlyDefaultAliases()));
+                                       $hashes[$relativeFileName] = hash('sha512', $oldMimetypeList->generateFile($this->mimeTypeDetector->getOnlyDefaultAliases(), false));
                                        continue;
                                }
                        }
```

**Comment by iasdeoupxe** [comment]
Confirmed on a Nextcloud 31.0.9 installation as well (having an own `config/mimetypemapping.json` file but no theme or `config/mimetypealiases.json` file in place).

Steps to reproduce:
1. Have a working installation without an  `INVALID_HASH` 
2. Run `php occ integrity:check-core`  and see no error / `INVALID_HASH`
3. Run both commands like described here: https://docs.nextcloud.com/server/stable/admin_manual/configuration_mimetypes/index.html
4. Run `php occ integrity:check-core` and see an `INVALID_HASH` for `core/js/mimetypelist.js`

If i'm making a diff between the current `core/js/mimetypelist.js` and https://raw.githubusercontent.com/nextcloud/server/refs/tags/v31.0.9/core/js/mimetypelist.js of my running version i can see the following diff:

```patch
--- <unnamed>
+++ <unnamed>
@@ -81,7 +81,6 @@
     "application/x-7z-compressed": "package/x-generic",
     "application/x-bzip2": "package/x-generic",
     "application/x-cbr": "text",
-    "application/x-compressed": "package/x-generic",
     "application/x-dcraw": "image",
     "application/x-deb": "package/x-generic",
     "application/x-fictionbook+xml": "text",
@@ -115,6 +114,7 @@
     "text/x-h": "text/code",
     "text/x-java-source": "text/code",
     "text/x-ldif": "text/code",
+    "text/x-nfo": "text/code",
     "text/x-python": "text/code",
     "text/x-rst": "text",
     "text/x-shellscript": "text/code",
```

Base is the is the original one from 31.0.9 and after running the update js commands one entry got dropped (`-`) and one added (`+`).

**Comment by hjeverts** [comment]
Same issue on 33.0.2 when updating from older Nextcloud verion

Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js
	- EXTRA_FILE
		- core/img/filetypes/drawio.svg
		- core/img/filetypes/dwb.svg

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a
                            [current] => 09779c15c6ca51e1dd07931ef2708756cbdc9a9adb41cd6a12eb177f57e005652e1ed3e1df18d46f80a4ad30637a6075e7a9cc11f34b44aeb66b3631b8ba1840
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [core/img/filetypes/drawio.svg] => Array
                        (
                            [expected] => 
                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
                        )

                    [core/img/filetypes/dwb.svg] => Array
                        (
                            [expected] => 
                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
                        )

                )

        )

)

**Comment by BloodyIron** [comment]
I just upgraded to v33.0.2 and I'm getting both the extra files drawio.svg and dwb.svg, which I deleted, but I also have the invalid hash of core/js/mimetypelist.js , looks like this was somehow missed in testing?

**Comment by logz254** [comment]
> Same issue on 33.0.2 when updating from older Nextcloud verion
> 
> # Technical information
> The following list covers which files have failed the integrity check. Please read the previous linked documentation to learn more about the errors and how to fix them.
> 
> # Results
> * core
>   
>   * INVALID_HASH
>     
>     * core/js/mimetypelist.js
>   * EXTRA_FILE
>     
>     * core/img/filetypes/drawio.svg
>     * core/img/filetypes/dwb.svg
> 
> # Raw output
> Array ( [core] => Array ( [INVALID_HASH] => Array ( [core/js/mimetypelist.js] => Array ( [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a [current] => 09779c15c6ca51e1dd07931ef2708756cbdc9a9adb41cd6a12eb177f57e005652e1ed3e1df18d46f80a4ad30637a6075e7a9cc11f34b44aeb66b3631b8ba1840 )
> 
> ```
>             )
> 
>         [EXTRA_FILE] => Array
>             (
>                 [core/img/filetypes/drawio.svg] => Array
>                     (
>                         [expected] => 
>                         [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
>                     )
> 
>                 [core/img/filetypes/dwb.svg] => Array
>                     (
>                         [expected] => 
>                         [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
>                     )
> 
>             )
> 
>     )
> ```
> 
> )

Same issue on v33.0.3

**Comment by gregecslo** [comment]
Same here, 33.0.3 just upgraded today.
It was same since V33...

Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js
	- EXTRA_FILE
		- core/img/filetypes/drawio.svg
		- core/img/filetypes/dwb.svg

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a
                            [current] => 09779c15c6ca51e1dd07931ef2708756cbdc9a9adb41cd6a12eb177f57e005652e1ed3e1df18d46f80a4ad30637a6075e7a9cc11f34b44aeb66b3631b8ba1840
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [core/img/filetypes/drawio.svg] => Array
                        (
                            [expected] => 
                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
                        )

                    [core/img/filetypes/dwb.svg] => Array
                        (
                            [expected] => 
                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
                        )

                )

        )

)

**Comment by iasdeoupxe** [comment]
@logz254 @gregecslo @hjeverts @BloodyIron Did any of you ran `occ maintenance:mimetype:update-js` at some point? If not you might need need to open a new issue as it could be unrelated to this here in that case.

**Comment by lightlike** [comment]
> [@logz254](https://github.com/logz254) [@gregecslo](https://github.com/gregecslo) [@hjeverts](https://github.com/hjeverts) [@BloodyIron](https://github.com/BloodyIron) Did any of you ran `occ maintenance:mimetype:update-js` at some point? If not you might need need to open a new issue as it could be unrelated to this here in that case.

That `occ` command was the thing that caused this issue to surface. At least for me.
I also just ran it and the integrity check again right now and it is still present.
The svg entries are now gone but I do not know if that is connected.

Mine still shows the following:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a
                            [current] => 6b290ba45e633706e1fe761ca21422053c113f17b2bb35f2351f2cf1c3b893aa543af6c594077ca213c755cfa0d002e22109bb1797bbadd8fb410ca839143e0e
                        )

                )

        )

)
```

**Comment by logz254** [comment]
> [@logz254](https://github.com/logz254) [@gregecslo](https://github.com/gregecslo) [@hjeverts](https://github.com/hjeverts) [@BloodyIron](https://github.com/BloodyIron) Did any of you ran `occ maintenance:mimetype:update-js` at some point? If not you might need need to open a new issue as it could be unrelated to this here in that case.

Yes and I am getting the same error as @lightlike

**Comment by hjeverts** [comment]
> [@logz254](https://github.com/logz254) [@gregecslo](https://github.com/gregecslo) [@hjeverts](https://github.com/hjeverts) [@BloodyIron](https://github.com/BloodyIron) Did any of you ran `occ maintenance:mimetype:update-js` at some point? If not you might need need to open a new issue as it could be unrelated to this here in that case.

Yes, I did.

**Comment by gregecslo** [comment]
sudo -u www-data php occ maintenance:mimetype:update-js

RESCAN



Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js
	- EXTRA_FILE
		- core/img/filetypes/drawio.svg
		- core/img/filetypes/dwb.svg

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a
                            [current] => 09779c15c6ca51e1dd07931ef2708756cbdc9a9adb41cd6a12eb177f57e005652e1ed3e1df18d46f80a4ad30637a6075e7a9cc11f34b44aeb66b3631b8ba1840
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [core/img/filetypes/drawio.svg] => Array
                        (
                            [expected] => 
                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
                        )

                    [core/img/filetypes/dwb.svg] => Array
                        (
                            [expected] => 
                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
                        )

                )

        )

)

**Comment by gregecslo** [comment]
rm core/img/filetypes/drawio.svg
rm core/img/filetypes/drawio.svg
sudo -u www-data php occ maintenance:mimetype:update-js

RESCAN

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => cb945c6402e12d9e7d42d0359acf95a6e9a9b0c1f3bd8528f598a7fb1694e5ae34c80cf44ef6c8901eac1bfdd152de3315fc7eac007efee0f33f09ed3e518b6a
                            [current] => 6b290ba45e633706e1fe761ca21422053c113f17b2bb35f2351f2cf1c3b893aa543af6c594077ca213c755cfa0d002e22109bb1797bbadd8fb410ca839143e0e
                        )

                )

        )

)

**Comment by iasdeoupxe** [comment]
Ok, then the `mimetypelist.js` looks the same underlying problem. The `drawio.svg` looks unrelated though and might require a separate issue 👍

**Comment by iasdeoupxe** [comment]
The ones having dwb.svg/drawio.svg results included as well might be still unrelated and could be an issue in the Draw.io app because that seems to inject additional files / changes to the Nextcloud server core code:

- https://github.com/jgraph/drawio-nextcloud/issues/70
- https://github.com/jgraph/drawio-nextcloud/issues/70#issuecomment-4366015025

And triggering the integrity failure in that case seems to be expected and not something the Nextcould server people could handle / fix?

**Comment by iasdeoupxe** [comment]
@gregecslo Deleting the files is not really a good idea because those seems be required / injected by the app linked previously.

**Comment by iasdeoupxe** [comment]
Anyone using just the Draw.io app, this seems to be deprecated according to https://github.com/jgraph/drawio-nextcloud/issues/70#issuecomment-4414927881 and https://github.com/jgraph/drawio-nextcloud/blob/main/README.md:

> This app is deprecated and will not receive further updates. Do not install it. If you are currently using it, please migrate away.

and should be removed from your installation to solve this.

**Comment by doc75** [comment]
I just tested the following:
 
1. Test 01

    - Upgrade for NC32 to NC33 with drawio installed
    - Result: integrity check error
    - removing drawio app
    - Result: integrity check error

2. Test 02

    - Uninstall drawio in NC32 
    - Upgrade for NC32 to NC33 with drawio installed
    - Result: integrity check OK (no error)

So definitely drawio is the guilty app

**Comment by iasdeoupxe** [comment]
So TL;DR:
1. One cause of issue is the draw.io app which is tracked on https://github.com/jgraph/drawio-nextcloud/issues/70
2. At least i never had the draw.io app enabled but still had this issue (https://github.com/nextcloud/server/issues/44315#issuecomment-3438842615) and for this case underlying issue still needs be found

**Comment by Ocres1** [comment]
Just want to confirm, the issue still exists on 34.0.0. When upgrading from 33 and an installed draw.io app installation the integrity check still fails. Removing rm core/img/filetypes/drawio.svg and core/img/filetypes/dwb.svg just resolves the failure for these, however for core/js/mimetypelist.js the check still keeps failing. Taking core the entire core directory from the repo it works, but after running occ maintenance:mimetype:update-js obviously the file gets modified and the check fails again. There must be fragements around.

A diff on mimetypelist.js shows:

29a30
>     "application/toml": "text/code",
94a96
>     "application/x-openvpn-profile": "text",
104a107,114
>     "audio/aac": "audio",
>     "audio/midi": "audio",
>     "audio/mpeg": "audio",
>     "audio/ogg": "audio",
>     "audio/opus": "audio",
>     "audio/wav": "audio",
>     "audio/webm": "audio",
>     "audio/x-midi": "audio",
120a131,134
>     "video/mp4": "video",
>     "video/mpeg": "video",
>     "video/ogg": "video",
>     "video/webm": "video",
171a186
> 'application/toml': t('core', "TOML configuration file"),
218a234
> 'application/x-openvpn-profile': t('core', "OpenVPN connection profile"),
 
Could anybody please explain how to fix or further investigate? 
Thanks!

**Comment by iasdeoupxe** [comment]
> Just want to confirm, the issue still exists on 34.0.0. When upgrading from 33 and an installed draw.io app installation the integrity check still fails.

Which is expected as devs of the draw.io app not fixed this (see one comment previously).

Also:

> Removing rm core/img/filetypes/drawio.svg and core/img/filetypes/dwb.svg just resolves the failure for these

Deleting file which got placed by installed app is not that a good idea :-)

> Could anybody please explain how to fix or further investigate?

According to https://github.com/jgraph/drawio-nextcloud/issues/70 original app developer/maintainer didn't intended to fix  but maybe new one is able to do:

https://github.com/arnowelzel/drawio-nextcloud/issues/9

Short: draw.io relevant integrity message probably needs be fixed in app and monitoring of linked app issue could be done.

**Comment by vinkin56** [comment]
Hi everyone!

Here’s how to fix it; I’ve done this on versions 28 and 33:
Let's use version 33.0.9 as an example.

1. Download the file: `wget https://download.nextcloud.com/server/releases/nextcloud-33.0.9.zip -O nextcloud-33.0.9.zip`

2. Extract the specific file: `unzip -j nextcloud-33.0.9.zip "nextcloud/core/js/mimetypelist.js" -d /tmp/`

3. Copy the file, overwriting the existing one (I’m using Docker; the principle is the same for host installations): `docker cp /tmp/mimetypelist.js nextcloud:/var/www/html/core/js/mimetypelist.js`

4. Set the file permissions: `docker exec --user root nextcloud chown www-data:www-data /var/www/html/core/js/mimetypelist.js`

5. Run a check: `docker exec --user www-data nextcloud php occ integrity:check-core`

6. Done—we’ve fixed it!
